From 107a0f1d4bb5e9a260e5db58a198703324eefe2d Mon Sep 17 00:00:00 2001
From: WANG Xuerui <xen0n@gentoo.org>
Date: Thu, 20 Feb 2025 09:45:32 +0000
Subject: [PATCH 5/8] Bug 1778177 - Enable LSX for loongarch64 libvpx.
 r=chunmin DONTBUILD

Differential Revision: https://phabricator.services.mozilla.com/D238350

(cherry picked from commit 8b8a1456e6984cc87e3833b807c6f486fc5b66f1)
---
 media/libvpx/moz.build | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/media/libvpx/moz.build b/media/libvpx/moz.build
index df29c9aae8838..846d946b64b36 100644
--- a/media/libvpx/moz.build
+++ b/media/libvpx/moz.build
@@ -87,6 +87,12 @@ elif CONFIG['TARGET_CPU'] == 'aarch64':
         SOURCES += files['LINUX_ARM64_SOURCES']
         ASFLAGS += [ '-I%s/media/libvpx/config/linux/arm64/' % TOPSRCDIR ]
         LOCAL_INCLUDES += [ '/media/libvpx/config/linux/arm64/' ]
+elif CONFIG['TARGET_CPU'] == 'loongarch64':
+    # LoongArch64 is Linux-only for now.
+    EXPORTS.vpx += files['LINUX_LOONGARCH64_EXPORTS']
+    SOURCES += files['LINUX_LOONGARCH64_SOURCES']
+    ASFLAGS += [ '-I%s/media/libvpx/config/linux/loongarch64/' % TOPSRCDIR ]
+    LOCAL_INCLUDES += [ '/media/libvpx/config/linux/loongarch64/' ]
 else:
     # Generic C-only configuration
     EXPORTS.vpx += files['GENERIC_EXPORTS']
@@ -133,6 +139,8 @@ for f in SOURCES:
             SOURCES[f].flags += ['-march=armv8.2-a+dotprod+i8mm+sve']
         if 'sve2.c' in f:
             SOURCES[f].flags += ['-march=armv9-a+sve2']
+        if 'lsx.c' in f:
+            SOURCES[f].flags += ['-mlsx']
 
 # Suppress warnings in third-party code.
 CFLAGS += [
-- 
2.48.1

